File: docs/examples/user_types/nodejs/01-userClasses.js
// classes-start
const { Oso } = require('oso');

const oso = new Oso();

class Customer {
  constructor(id) {
    this.id = id;
  }
}

oso.registerClass(Customer);

class InternalUser {
  constructor(id) {
    this.id = id;
  }
}

oso.registerClass(InternalUser);
// classes-end

// app-start
async function customerDashboardHandler(request) {
  const actor = userFromId(request.id);
  return oso.isAllowed(actor, 'view', 'customer_dashboard');
}

function userFromId(id) {
  const userType = db.query('SELECT type FROM users WHERE id = ?', id);
  if (userType === 'internal') {
    return new InternalUser(id);
  } else if (userType === 'customer') {
    return new Customer(id);
  }
}
// app-end

File: docs/examples/user_types/python/01-user_classes.py
class Customer:
    def __init__(self, id):
        self.id = id


class InternalUser:
    def __init__(self, id):
        self.id = id
        # classes-end


# app-start
def customer_dashboard_handler(request):
    oso = get_oso()
    actor = user_from_id(request.id)
    allowed = oso.is_allowed(actor=actor, action="view", resource="customer_dashboard")


def user_from_id(id):
    user_type = db.query("SELECT type FROM users WHERE id = ?", id)
    if user_type == "internal":
        return InternalUser(id)
    elif user_type == "customer":
        return Customer(id)
        # app-end
